spec/message.abnf

```diff
@@ -52,16 +55,18 @@ unquoted-start = name-start / DIGIT / "."
                / %xB7 / %x300-36F / %x203F-2040
 
 
-; reserve sigils for private-use by implementations
-private-use    = private-start reserved-body
-private-start  = "^" / "&"
+; Reserve additional .keywords for use by future versions of this specification.
+; Note that the following expression is a simplification,
+; as this rule MUST NOT be considered to match existing keywords:
+;   (input / local / match) (s / "{")
+reserved-statement = "." reserved-body 1*expression
```

Comment:
I think requiring `reserved-statement` to end with at least one `expression` matches too little.
```suggestion
; A reserved statement continues until the next statement or quoted-pattern,
; and may include any number of expressions and/or quoted literals
reserved-statement      = "." *([s] reserved-statement-part)
reserved-statement-part   = 1*(reserved-statement-char / reserved-statement-escape)
                          / expression
                          / quoted
reserved-statement-char   = %x00-08        ; omit `s`: HTAB and LF
                          / %x0B-0C        ; omit `s`: CR
                          / %x0E-19        ; omit `s`: SP
                          / %x21-2D        ; omit .
                          / %x2F-5B        ; omit \
                          / %x5D-7A        ; omit { | }
                          / %x7E-D7FF      ; omit surrogates
                          / %xE000-10FFFF
reserved-statement-escape = backslash ( %x00-D7FF / %xE000-10FFFF )
```
Note that this locks in the `expression` syntax for effectively forever because a left brace _could_ start a `{{…}}` quoted expression which would terminate the reserved statement—though possible to work around that too; it's rather impractical. And likewise  for `quoted`.

Reply:
The `.` exclusion also disallows using e.g. `name` within a future statement, so e.g. introducing `local-declaration` would not be possible if we didn't include it already.

On the other hand, if each `declaration` had to end in a required whitespace `s`, we would only need to bar the sequence `(s ".")` from what's reserved.

Reply:
@gibson042 Please correct me if I misremember, but my understanding from today is that we're fine with reserving only statements that end in an expression?